lib/config.js

```diff
@@ -40,28 +40,42 @@ module.exports = (json) => {
       const bin = contract["bin"]
       const bin_runtime = contract["bin-runtime"]
       const srcmap_runtime = contract["srcmap-runtime"]
+      const srcmap = contract["srcmap"]
 
       if (!ast)            warn(`Contract AST not found for ${fullname}`)
       if (!abi)            warn(`Contract abi not found for ${fullname}`)
       if (!bin)            warn(`Contract bin not found for ${fullname}`)
       if (!bin_runtime)    warn(`Contract bin_runtime not found for ${fullname}`)
       if (!srcmap_runtime) warn(`Contract srcmap_runtime not found for ${fullname}`)
+      if (!srcmap) warn(`Contract srcmap not found for ${fullname}`)
 
-      const { instructions, pc_to_inst_map } = get_pc_to_inst_map(bin_runtime)
-      const inst_to_pc = instructions.map((inst, i) => pc_to_inst_map.indexOf(i));
-      const srcmapArr = genSrcmapArr(srcmap_runtime);
+      const bin_instructions_meta = get_pc_to_inst_map(bin)
+      const bin_instructions = bin_instructions_meta.instructions;
+      const bin_pc_to_inst_map = bin_instructions_meta.pc_to_inst_map;
+      const bin_runtime_instructions_meta = get_pc_to_inst_map(bin_runtime)
+      const bin_runtime_instructions = bin_runtime_instructions_meta.instructions;
+      const bin_runtime_pc_to_inst_map = bin_runtime_instructions_meta.pc_to_inst_map;
+      const bin_inst_to_pc = bin_instructions.map((inst, i) => bin_pc_to_inst_map.indexOf(i));
+      const bin_runtime_inst_to_pc = bin_runtime_instructions.map((inst, i) => bin_runtime_pc_to_inst_map.indexOf(i));
+      const bin_runtime_srcmapArr = genSrcmapArr(srcmap_runtime);
+      const bin_srcmapArr = genSrcmapArr(srcmap_runtime);
```

Comment:
What is the distinction between `bin` and `bin_runtime` here? Is `bin` the creation code?

Reply:
Yes `bin_runtime` is the program code and `bin` is the creation code (`INITCODE` in the evm semantics)

Reply:
maybe `bin_init` or `bin_creation` would be a little clearer?

Reply:
`bin` and `bin_runtime` are what solidity calls it (you see it in the flag `--combined-json=bin-runtime` etc.) and in the soldity JSON output. So I think it makes sense to stick with these names.

Reply:
did not know that, make sense.